repose-aggregator/components/services/container-configuration-service/impl/src/main/scala/org/openrepose/nodeservice/containerconfiguration/ContainerConfigurationServiceImpl.scala

```diff
@@ -0,0 +1,146 @@
+    configurationService.unsubscribeFrom(ContainerConfigurationFilename, this)
+  }
+
+  override def getVia: Optional[String] = {
+    initializationCheck()
+    Optional.ofNullable(patchedDeploymentConfiguration.get().getVia)
+  }
+
+  override def getContentBodyReadLimit: Optional[Long] = {
+    initializationCheck()
+    Optional.ofNullable(patchedDeploymentConfiguration.get().getContentBodyReadLimit)
+  }
+
+  override def getDeploymentConfiguration: DeploymentConfiguration = {
+    initializationCheck()
+    patchedDeploymentConfiguration.get()
+  }
+
+  override def subscribeTo(listener: UpdateListener[DeploymentConfiguration]): Unit = {
+    subscribeTo(listener, sendNotificationNow = true)
+  }
+
+  override def subscribeTo(listener: UpdateListener[DeploymentConfiguration], sendNotificationNow: Boolean): Unit = {
+    initializationCheck()
+    logger.debug("Subscribing listener with hash code: {}", listener.hashCode.toString)
+
+    // TODO: Clean this up with Scala 2.12 support for Scala function -> Java 8 function
+    updateListeners.updateAndGet(new UnaryOperator[Set[UpdateListener[DeploymentConfiguration]]] {
+      override def apply(t: Set[UpdateListener[DeploymentConfiguration]]): Set[UpdateListener[DeploymentConfiguration]] =
+        t + listener
+    })
+
+    if (sendNotificationNow) listener.configurationUpdated(patchedDeploymentConfiguration.get())
+  }
+
+  override def unsubscribeFrom(listener: UpdateListener[DeploymentConfiguration]): Unit = {
+    initializationCheck()
+    logger.debug("Unsubscribing listener with hash code: {}", listener.hashCode.toString)
+
+    updateListeners.updateAndGet(new UnaryOperator[Set[UpdateListener[DeploymentConfiguration]]] {
+      override def apply(t: Set[UpdateListener[DeploymentConfiguration]]): Set[UpdateListener[DeploymentConfiguration]] =
+        t - listener
+    })
+  }
+
+  override def isInitialized: Boolean =
+    initialized.get()
+
+  override def configurationUpdated(containerConfiguration: ContainerConfiguration): Unit = {
+    val baseConfig = containerConfiguration.getDeploymentConfig
+    val patchedDeploymentConfig = containerConfiguration.getClusterConfig
+      .find(patchConfig => clusterId.equals(patchConfig.getClusterId))
+      .map(patchConfig => DeploymentConfigPatchUtil.patch(baseConfig, patchConfig))
+      .getOrElse(baseConfig)
+
+    patchedDeploymentConfiguration.set(patchedDeploymentConfig)
+    initialized.set(true)
+
+    updateListeners.get() foreach { listener =>
+      try {
+        listener.configurationUpdated(patchedDeploymentConfig)
+      } catch {
+        case ex: Exception =>
+          logger.error("Configuration update error. Reason: {}", ex.getLocalizedMessage)
+          logger.trace("", ex)
+      }
+    }
+  }
+
+  // Note: The container configuration should be read on init of this class. As such, this check
+  //       should not ever fail.
+  private def initializationCheck(): Unit =
```

Comment:
Doing this like this, if the inits are called out of order, those components that init before will never be subscribed. You should probably double check in the spring documentation when `PostConstruct` is called, and if there is any guarantee on order.